# HG changeset patch
# User pantxo <pantxo.diribarne@gmail.com>
# Date 1402934631 25200
#      Mon Jun 16 09:03:51 2014 -0700
# Branch stable
# Node ID 7bf70dc7f06c7eb11c5fafdfd97b501cd56e5e78
# Parent  94d791a0373727c0501c87a952d144ff94b5020c
Correctly return 0 for empty matrices (bug #42563).

* xnorm.cc (xnorm): Add check for empty matrix and return 0 if found.

diff --git a/libinterp/corefcn/xnorm.cc b/libinterp/corefcn/xnorm.cc
--- a/libinterp/corefcn/xnorm.cc
+++ b/libinterp/corefcn/xnorm.cc
@@ -43,17 +43,19 @@ octave_value xnorm (const octave_value& 
 
   bool isvector = (x.columns () == 1 || x.rows () == 1);
   bool iscomplex = x.is_complex_type ();
   bool issparse = x.is_sparse_type ();
   bool isfloat = x.is_single_type ();
 
   if (isfloat || x.is_double_type ())
     {
-      if (isvector)
+      if (x.is_empty ())
+        retval = octave_value (0);
+      else if (isvector)
         {
           if (isfloat & iscomplex)
             retval = xnorm (x.float_complex_column_vector_value (),
                             p.float_value ());
           else if (isfloat)
             retval = xnorm (x.float_column_vector_value (),
                             p.float_value ());
           else if (iscomplex)
